### PR TITLE
SOC-755 Set tracking for /join page

### PR DESCRIPTION
**Pull request by bkoval**
@kvas-damian @mklucsarits If SOC-753 is merged to dev I'll retarget it to dev.

https://wikia-inc.atlassian.net/browse/SOC-755

### CODE REVIEW

**Comment by rogatty** [comment]
```
front/scripts/auth/tracking.ts(21,10): 2322 Type 'Element' is not assignable to type 'boolean'.
Build contains some typescript errors/warnings
+ echo BUILD:FAILURE
```

File: front/scripts/auth/tracking.ts [hunk 1]
```diff
@@ -16,6 +16,11 @@
 		}
 	});
 
+	function isJoinPage(): boolean {
```
Comment (mklucsarits) [inline front/scripts/auth/tracking.ts]: Please add a space between the function name and parentheses for all the functions here. It's been a convention for the Mercury codebase.

Comment (mklucsarits) [inline front/scripts/auth/tracking.ts]: Also, this does not return a boolean. Prefix with ```!!``` if you need a bool here, or change the return type.

File: front/scripts/auth/tracking.ts [hunk 2]
```diff
@@ -27,44 +32,88 @@
...
+				action: Mercury.Utils.trackActions.close,
+				category: 'user-login-mobile',
+				label: 'join-close-button'
+			});
+		});
+	}
+
+	function init(): void {
+		if (isJoinPage()) {
+			setTrackingForJoinPage()
```
Comment (mklucsarits) [inline front/scripts/auth/tracking.ts]: Missing semi-colon.

File: front/scripts/auth/tracking.ts [hunk 3]
```diff
@@ -27,44 +32,88 @@
...
-	// Click "Sign In" button
-	document.getElementById('loginSubmit').addEventListener('click', (): void => {
-		Mercury.Utils.track({
-			trackingMethod: 'ga',
-			action: Mercury.Utils.trackActions.click,
-			category: 'user-login-mobile',
-			label: 'login-submit'
+	function setTrackingForLoginPage(): void {
+		// Click "Sign In" button
+		document.getElementById('loginSubmit').addEventListener('click', (): void => {
```
Comment (kvas-damian) [inline front/scripts/auth/tracking.ts]: What about forms that are submited by hitting enter - not submit button?

File: front/scripts/auth/tracking.ts [hunk 4]
```diff
@@ -27,44 +32,88 @@
...
+			M.track({
+				trackingMethod: 'ga',
```
Comment (kvas-damian) [inline front/scripts/auth/tracking.ts]: I'd say that tracking in GA is not enough. What do you think about ```both```?

File: front/scripts/auth/tracking.ts [hunk 5]
```diff
@@ -27,44 +32,88 @@
...
-	// Click X to "close" log-in form
-	document.querySelector('.close').addEventListener('click', (): void => {
-		Mercury.Utils.track({
-			trackingMethod: 'ga',
-			action: Mercury.Utils.trackActions.close,
-			category: 'user-login-mobile',
-			label: 'login-modal'
+		// Click X to "close" log-in form
+		document.querySelector('.close').addEventListener('click', (): void => {
+			M.track({
```
Comment (kvas-damian) [inline front/scripts/auth/tracking.ts]: I see a lot of calls to ```M.track()``` and some params are always the same, what do you think about creating local wrapper?

File: front/scripts/auth/tracking.ts [hunk 6]
```diff
@@ -27,44 +32,88 @@
...
-			trackingMethod: 'ga',
-			action: Mercury.Utils.trackActions.click,
-			category: 'user-login-mobile',
-			label: 'login-submit'
+	function setTrackingForLoginPage(): void {
+		// Click "Sign In" button
+		document.getElementById('loginSubmit').addEventListener('click', (): void => {
+			M.track({
+				trackingMethod: 'ga',
+				action: Mercury.Utils.trackActions.click,
```
Comment (kvas-damian) [inline front/scripts/auth/tracking.ts]: Maybe we could use ```submit``` here?

File: front/scripts/auth/tracking.ts (hunk 3 above)
Comment (bkoval) [inline front/scripts/auth/tracking.ts]: Not my code, but good catch. I'll use the submit event here :)

**Comment by bkoval** [comment]
Closing, since the new pull request will target dev.

**Pull request closed without merging**